docs/guide/migration.md

```diff
@@ -207,6 +207,8 @@ There are some changes which only affect plugin/tool creators.
   - The `configurePreviewServer` hook now accepts the `PreviewServer` type instead of `PreviewServerForHook` type.
 - [[#14818] refactor(preview)!: use base middleware](https://github.com/vitejs/vite/pull/14818)
   - Middlewares added from the returned function in `configurePreviewServer` now does not have access to the `base` when comparing the `req.url` value. This aligns the behaviour with the dev server. You can check the `base` from the `configResolved` hook if needed.
+- [[#14834] fix(types)!: expose httpServer with Http2SecureServer union](https://github.com/vitejs/vite/pull/14834)
+  - `http.Server | http2.Http2SecureServer` type is now used instead of `http.Server` type where appropriate.
```

Comment:
I think it should be "the `http.Server` type", but maybe we could remove the word type? It should be ok given the name of the commit. 
```suggestion
  - `http.Server | http2.Http2SecureServer` is now used instead of `http.Server` where appropriate.
```